src/NSwag.SwaggerGeneration.AspNetCore.Tests/AspNetCoreToSwaggerGenerationTests.cs

```diff
@@ -0,0 +1,446 @@
+        {
+            [HttpGet("/test")]
+            public Task<TestModel> FindModel([FromRoute] int id) => null;
+        }
+
+        private class ActionWithSwaggerIgnoreAttribute
+        {
+            [HttpGet("/test1")]
+            public Task<TestModel> PublicApi() => null;
+
+            [SwaggerIgnore]
+            [HttpGet("/test2")]
+            public Task<TestModel> SecretApi([FromRoute] int id) => null;
+        }
+
+        private class ParameterWithSwaggerIgnoreAttribute
+        {
+            [HttpPost("{id}/{name?}")]
+            public Task<TestModel> GetModel([FromRoute] int id, [FromRoute] [SwaggerIgnore] string name) => null;
+        }
+
+        [Route("[controller]/[action]")]
+        public class HttpMethodsController
+        {
+            [AcceptVerbs("Delete")]
+            public void HttpDelete() { }
+
+            [AcceptVerbs("pAtch")]
+            public void HttpPatch() { }
+
+            [HttpHead]
+            public void HttpHead() { }
+
+            [AcceptVerbs("OPTIONS")]
+            public void HttpOptions() { }
+        }
+
+        public class ActionWithSwaggerOperationAttribute
+        {
+            [HttpGet("/test")]
+            [SwaggerOperation("CustomOperationId")]
+            public void TestAction() { }
+        }
+
+        [SwaggerOperationProcessor(typeof(TestSwaggerOperationProcessor), "Hello from controller")]
+        public class ControllerWithSwaggerOperationProcessor
+        {
+            [HttpGet("/test")]
+            public void TestAction() { }
+        }
+
+        public class ActionWithSwaggerOperationProcessor
+        {
+            [SwaggerOperationProcessor(typeof(TestSwaggerOperationProcessor), "Hello from action")]
+            [HttpGet("/test")]
+            public void TestAction() { }
+        }
+
+        private class TestSwaggerOperationProcessor : IOperationProcessor
+        {
+            public TestSwaggerOperationProcessor(string value)
+            {
+                Value = value;
+            }
+
+            public string Value { get; }
+
+            public Task<bool> ProcessAsync(OperationProcessorContext context)
+            {
+                context.Document.Info.Title = Value;
+                return Task.FromResult(true);
+            }
+        }
+
+        [SwaggerResponse(202, typeof(TestModel))]
+        private class ControllerWithSwaggerResponseTypeAttribute
+        {
+            [HttpPut("/test1")]
+            public IActionResult UpdateModel(int id) => null;
+        }
+
+        private class ActionWithSwaggerResponseAttribute
+        {
+            [SwaggerResponse(201, typeof(TestModel))]
+            [HttpPut("/test1")]
+            public IActionResult CreateModel(int id) => null;
+        }
+
+        private class TestModel { }
+
+        #endregion
+
+        #region ApiDescription generation
+
+        private static ApiDescriptionGroupCollection GetApiDescriptionGroups(Type controllerType)
```

Comment:
This should have some comments to explain what it's for and why it's here.

You can use this approach to cheaply get MVC to do all of the initial reflection for you, but just be aware that this isn't guaranteed to return up-to-date results - it will always be locked to 1.0.0-s behavior.

We also need good unit tests that construct `ApiDescription` directly without this.

Reply:
> We also need good unit tests that construct ApiDescription directly without this.

Sure